### PR TITLE
feat: create wallet

### PR DESCRIPTION
**Pull request by guanbinrui**
+ Add `<CreateWalletDialog />` to implement ([figma](https://www.figma.com/file/gVkQ67y285b4FXVV1KPThN/Twitter?node-id=7655%3A223712)).
+ All wallets should be created manually.

## Screenshots

image
image
image

### CODE REVIEW

File: packages/maskbook/src/plugins/Wallet/hooks/useMnemonicWordsPuzzle.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    )
+
+    const resetCallback = useCallback(() => {
+        setAnswerWords([])
+        setSeed((x) => (x + 1) % 3)
+    }, [])
+
+    const refreshCallback = useCallback(() => {
+        wordsRetry()
+    }, [wordsRetry])
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Wallet/hooks/useMnemonicWordsPuzzle.ts:48]: So why not just `wordsRetry`

**Review by Jack-Works — APPROVED** [review]

**Comment by Jack-Works** [comment]
is this ready for review?

**Comment by Jack-Works** [comment]
mark it as draft. cc me if it's ready for review! @guanbinrui

**Comment by Jack-Works** [comment]
here is merge conflict @guanbinrui

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/plugins/Wallet/database/Wallet.db.ts [hunk 2]
```diff
@@ -46,24 +30,19 @@ export const createWalletDBAccess = createDBAccess(() => {
                 // @ts-ignore
                 os.createIndex('2', 'index2')
                 os.createIndex('plugin_id', 'plugin_id')
-                // @ts-ignore
-                db.deleteObjectStore('GitcoinDonation')
-                // @ts-ignore
-                db.deleteObjectStore('RedPacket')
```
[the remark this answers is not in the document]
Comment (guanbinrui) [reply packages/maskbook/src/plugins/Wallet/database/Wallet.db.ts:50]: It's a tiny part of users and nothing harmful if that happens.

File: packages/maskbook/src/plugins/Wallet/database/Wallet.db.ts [hunk 3]
```diff
@@ -14,29 +15,12 @@ function path<T>(x: T) {
     return x
 }
 export const createWalletDBAccess = createDBAccess(() => {
-    return openDB<WalletDB>('maskbook-plugin-wallet', 6, {
+    return openDB<WalletDB>('maskbook-plugin-wallet', 7, {
         async upgrade(db, oldVersion, newVersion, tx) {
             function v0_v1() {
-                // @ts-expect-error
-                db.createObjectStore('RedPacket', { keyPath: path<keyof RedPacketRecordInDatabase>('id') })
```
[the remark this answers is not in the document]
Comment (guanbinrui) [reply packages/maskbook/src/plugins/Wallet/database/Wallet.db.ts:18]: Yes, no necessary to create records that will delete immediately.

**Review by septs — APPROVED** [review]

File: packages/maskbook/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleCard.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+export function CollectibleCard(props: CollectibleCardProps) {
+    const { wallet, token } = props
+    const classes = useStyles(props)
+
+    return (
+        <Link target="_blank" rel="noopener noreferrer" href={props.link ?? ''}>
+            <Card className={classes.root} style={{ width: 160, height: 220 }}>
+                <ActionsBarNFT classes={{ more: classes.icon }} wallet={wallet} token={token} />
+                {token.image ? (
+                    <Image
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleCard.tsx:48]: Image is not necessary in the options page.

File: packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet.tsx [hunk 5]
```diff
@@ -747,3 +959,374 @@ export function DashboardWalletRedPacketDetailDialog(
...
+                            onClick={onTransfer}>
+                            {validationMessage || t('wallet_transfer_send')}
+                        </Button>
+                    </div>
+                }
+            />
+        </DashboardDialogCore>
+    )
+}
+//#endregion
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet.tsx:1332]: Don't you think this file is tooooooooo long?

**Review by Jack-Works — APPROVED** [review]
I roughly reviewed the style of this PR and generally LGTM.

Another thing to mention. It seems like you're using patches as a regular release which is not correct in the workflow. patch release should be only used for bug fixes. New features should use master branch as the base branch.

**Pull request merged**